llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -173,6 +171,38 @@ bool DataScalarizerVisitor::visitStoreInst(StoreInst &SI) {
   return false;
 }
 
+bool DataScalarizerVisitor::visitExtractElementInst(ExtractElementInst &EEI) {
```

Comment:
```suggestion
bool DataScalarizerVisitor::visitExtractElementInst(ExtractElementInst &EEI) {
  // Shader Model 6.9 and above supports dynamic vector indexing and therefore
  // does not need to be scalarized
  VersionTuple ShaderModelVersion =
      EEI.getModule()->getTargetTriple().getOSVersion();
  if (ShaderModelVersion >= VersionTuple(6, 9))
    return false;
    
```
Should I add a shader model check here for 6.9? SM 6.9 isn't currently a supported target. Perhaps a TODO comment instead?

Reply:
I don't know if this is needed we likely need to disable the entire data scalarization pass  for 6.9 not just this one behavior.

Reply:
I also don't like to prematurely assume correct behavior for something we haven't planned out. So would prefer we don't make any assumptions about 6.9.